tests/integration/sync/test_sync_command.py

```diff
@@ -0,0 +1,363 @@
+from tests.testing_utils import run_command_with_input
+
+# Deploy tests require credentials and CI/CD will only add credentials to the env if the PR is from the same repo.
+# This is to restrict package tests to run outside of CI/CD, when the branch is not master or tests are not run by Canary
+SKIP_SYNC_TESTS = RUNNING_ON_CI and RUNNING_TEST_FOR_MASTER_ON_CI and not RUN_BY_CANARY
+CFN_SLEEP = 3
+API_SLEEP = 5
+SFN_SLEEP = 10
+CFN_PYTHON_VERSION_SUFFIX = os.environ.get("PYTHON_VERSION", "0.0.0").replace(".", "-")
+
+LOG = logging.getLogger(__name__)
+
+
+@skipIf(SKIP_SYNC_TESTS, "Skip sync tests in CI/CD only")
+class TestSync(BuildIntegBase, SyncIntegBase, PackageIntegBase):
+    @classmethod
+    def setUpClass(cls):
+        # setup signing profile arn & name
+        cls.signing_profile_name = os.environ.get("AWS_SIGNING_PROFILE_NAME")
+        cls.signing_profile_version_arn = os.environ.get("AWS_SIGNING_PROFILE_VERSION_ARN")
+
+        PackageIntegBase.setUpClass()
+
+        cls.test_data_path = Path(__file__).resolve().parents[1].joinpath("testdata", "sync")
+
+    def setUp(self):
+        self.cfn_client = boto3.client("cloudformation")
+        self.ecr_client = boto3.client("ecr")
+        self.lambda_client = boto3.client("lambda")
+        self.api_client = boto3.client("apigateway")
+        self.sfn_client = boto3.client("stepfunctions")
+        self.sns_arn = os.environ.get("AWS_SNS")
+        self.stacks = []
+        time.sleep(CFN_SLEEP)
+        super().setUp()
+
+    def tearDown(self):
+        shutil.rmtree(os.path.join(os.getcwd(), ".aws-sam", "build"), ignore_errors=True)
+        for stack in self.stacks:
+            # because of the termination protection, do not delete aws-sam-cli-managed-default stack
+            stack_name = stack["name"]
+            if stack_name != SAM_CLI_STACK_NAME:
+                region = stack.get("region")
+                cfn_client = (
+                    self.cfn_client if not region else boto3.client("cloudformation", config=Config(region_name=region))
+                )
+                ecr_client = self.ecr_client if not region else boto3.client("ecr", config=Config(region_name=region))
+                self._delete_companion_stack(cfn_client, ecr_client, self._stack_name_to_companion_stack(stack_name))
+                cfn_client.delete_stack(StackName=stack_name)
+        super().tearDown()
+
+    @parameterized.expand(["ruby", "python"])
+    def test_sync_infra(self, runtime):
+        template_before = f"infra/template-{runtime}-before.yaml"
+        template_path = str(self.test_data_path.joinpath(template_before))
+        stack_name = self._method_to_stack_name(self.id())
+        self.stacks.append({"name": stack_name})
+
+        # Run infra sync
+        sync_command_list = self.get_sync_command_list(
+            template_file=template_path,
+            code=False,
+            watch=False,
+            resource_id=None,
+            resource=None,
```

Comment:
nit: can we clean up parameters which have `None` value (assuming default values are `None`)?